provider/spark.go

```diff
@@ -1722,31 +1722,40 @@ func blobSparkMaterialization(id ResourceID, spark *SparkOfflineStore, isUpdate
 	if err != nil {
 		return nil, fmt.Errorf("could not create empty filepath due to error %w (store type: %s; path: %s)", err, spark.Store.FilestoreType(), sparkResourceTable.schema.SourceTable)
 	}
-
-	err = sourcePath.ParseDirPath(sparkResourceTable.schema.SourceTable)
-	if err != nil {
-		return nil, fmt.Errorf("could not parse full path due to error %w (store type: %s; path: %s)", err, spark.Store.FilestoreType(), sparkResourceTable.schema.SourceTable)
-	}
-	spark.Logger.Debugw("Parsed source table path:", "sourceTablePath", sourcePath.ToURI(), "sourceTable", sparkResourceTable.schema.SourceTable)
-	// TODO: Refactor this into a separate method
-	sourceFiles, err := spark.Store.List(sourcePath, filestore.Parquet)
-	if err != nil {
-		return nil, fmt.Errorf("could not get latest source file: %v", err)
-	}
-	groups, err := filestore.NewFilePathGroup(sourceFiles, filestore.DateTimeDirectoryGrouping)
-	if err != nil {
-		return nil, fmt.Errorf("could not get datetime directory grouping for source files: %v", err)
-	}
-	newest, err := groups.GetFirst()
-	if err != nil {
-		return nil, fmt.Errorf("could not get newest source file: %v", err)
-	}
-	sourceUris := make([]string, len(newest))
-	for i, sourceFile := range newest {
-		sourceUris[i] = sourceFile.ToURI()
+	var sourceURIs []string
```

Comment:
yah this whole thing should be it's own method. GetSourceUrisFromPath or something

I'm not familiar with a lot of these functions, like NewFilePathGroup or even Groups. trusting ya